tests/Freckle/App/Memcached/ServersSpec.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE CPP #-}
+
+module Freckle.App.Memcached.ServersSpec
+  ( spec
+  ) where
+
+import Freckle.App.Prelude
+
+import Control.Error.Util (hush)
+import Data.Either (isLeft, isRight)
+import qualified Database.Memcache.Client as Memcache
+import Freckle.App.Memcached.Servers
+import Freckle.App.Test
+
+spec :: Spec
+spec = do
+  describe "readMemcachedServers" $ do
+    it "requires the correct prefix" $ example $ do
+      void (readMemcachedServers "http://") `shouldSatisfy` isLeft
+      void (readMemcachedServers "memcached://") `shouldSatisfy` isRight
+
+    it "treats empty cavlues as default" $ example $ do
+      readServerSpecs "" `shouldBe` Just [Memcache.def]
```

Comment:
Now that I've introduced `MemcacheClientDisabled` as a value, should this `MEMCACHED_SERVERS=` case result in that (vs `MEMCACHED_SERVERS=memcached://` to get "defaults")? It would make things slightly nigher over in deployment to unconditionally add the variable, but conditionally make it empty for the environments without caching.